src/init-ceph.in

```diff
@@ -1021,6 +1021,16 @@ EOF
 		# ceph processes answer in around 100ms when the process works correctly
 		do_cmd "timeout 1 $BINDIR/ceph --admin-daemon $asok version 2>/dev/null || echo unknown"
 
+        	# log ceph osd state
+		if [ "$type" = "osd" ];then
+		    CEPH_DAEMON_STATUS=""
+		    execute_ceph_cmd CEPH_DAEMON_STATUS $name "ceph daemon $name status"
+		    if [ $? -eq 0 ]; then
+		        state=$(echo "$CEPH_DAEMON_STATUS" | awk -F ':' '/state/{gsub(/,/, "", $2);print $2}')
+		        wlog $name "INFO" "$name has the following state: $state"
```

Comment:
I just validated this command on my installed system. I'm wondering if this will create a bunch of logs where most of the time the OSD will just report that it's in the "active" state.  

So:
 a) During initial boot/start of the OSD, do we see other states than "active" and what are they?
 b) It might be better to only log a message for non-"active" states so that for a long running system we won't roll the ceph logs as quickly (preserve a longer hostory)

Reply:
a) This code is executed only when do init-ceph.in status, so it will not output anything when we start the osd (with init-ceph.in start)

b) I agree, or maybe we can cache the state somehow and log it only when a change occurs